test/e2e/create_test.go

```diff
@@ -44,50 +45,114 @@ func testWindowsNodeCreation(t *testing.T) {
+	}
+	return nil
+}
 
-	t.Run("Windows Machines with the Windows label are configured", func(t *testing.T) {
-		_, err := testCtx.createWindowsMachineSet(gc.numberOfNodes, true)
-		require.NoError(t, err, "failed to create Windows MachineSet")
-		// We need to cover the case where a user changes the private key secret before the WMCO has a chance to
-		// configure the Machine. In order to simulate that case we need to wait for the MachineSet to be fully
-		// provisioned and then change the key. The the correct amount of nodes being configured is proof that the
-		// mismatched Machine created with the mismatched key was deleted and replaced.
-		// Depending on timing and configuration flakes this will either cause all Machines, or all Machines after
-		// the first configured Machines to hit this scenario. This is a platform agonistic test so we run it only on
-		// AWS.
-		err = testCtx.waitForWindowsMachines(int(gc.numberOfNodes), "Provisioned")
-		require.NoError(t, err, "error waiting for Windows Machines to be provisioned")
-		if testCtx.CloudProvider.GetType() == config.AWSPlatformType {
-			// Replace the known private key with a randomly generated one.
-			err = testCtx.createPrivateKeySecret(false)
-			require.NoError(t, err, "error replacing private key secret")
+// testMachineConfiguration tests that the Windows Machine controller can properly configure Machines
+func (tc *testContext) testMachineConfiguration(t *testing.T) {
+	if gc.numberOfMachineNodes == 0 {
+		t.Skip("Machine Controller testing disabled")
+	}
+	_, err := tc.createWindowsMachineSet(gc.numberOfMachineNodes, true)
+	require.NoError(t, err, "failed to create Windows MachineSet")
+	// We need to cover the case where a user changes the private key secret before the WMCO has a chance to
+	// configure the Machine. In order to simulate that case we need to wait for the MachineSet to be fully
+	// provisioned and then change the key. The the correct amount of nodes being configured is proof that the
+	// mismatched Machine created with the mismatched key was deleted and replaced.
+	// Depending on timing and configuration flakes this will either cause all Machines, or all Machines after
+	// the first configured Machines to hit this scenario. This is a platform agonistic test so we run it only on
+	// AWS.
+	_, err = tc.waitForWindowsMachines(int(gc.numberOfMachineNodes), "Provisioned", true)
+	require.NoError(t, err, "error waiting for Windows Machines to be provisioned")
+	if tc.CloudProvider.GetType() == config.AWSPlatformType {
+		// Replace the known private key with a randomly generated one.
+		err = tc.createPrivateKeySecret(false)
+		require.NoError(t, err, "error replacing private key secret")
+	}
+	err = tc.waitForWindowsNodes(gc.numberOfMachineNodes, false, false, false)
+	assert.NoError(t, err, "Windows node creation failed")
+
+	t.Run("Changing the private key ensures all Windows nodes use the new private key", func(t *testing.T) {
```

Comment:
Now that you are moving the code, please rename the test to something succinct. `Change private key` or something to that effect.